From 9c239e50f463e3b36257330cdfc58aa92e77d99a Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:43:10 -0400
Subject: [infiniband] cxgb4: Initialize reserved fields in a FW work request

Message-id: <1410475447-94392-52-git-send-email-svemuri@redhat.com>
Patchwork-id: 93837
O-Subject: [RHEL7.1 PATCH BZ 1124947 051/108] RDMA/cxgb4: Initialize reserved fields in a FW work request
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Signed-off-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit c3f98fa29176753a759ade424f18b11f440b19f4)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/cxgb4/qp.c b/drivers/infiniband/hw/cxgb4/qp.c
index 2c037e1..5a7d368 100644
--- a/drivers/infiniband/hw/cxgb4/qp.c
+++ b/drivers/infiniband/hw/cxgb4/qp.c
@@ -436,6 +436,8 @@ static int build_rdma_send(struct t4_sq *sq, union t4_wr *wqe,
  default:
   return -EINVAL;
  }
+ wqe->send.r3 = 0;
+ wqe->send.r4 = 0;
 
  plen = 0;
  if (wr->num_sge) {
-- 
1.7.1